staging/src/k8s.io/component-base/metrics/counter.go

```diff
@@ -107,26 +107,25 @@ func (c *Counter) initializeDeprecatedMetric() {
 
 // WithContext allows the normal Counter metric to pass in context.
 func (c *Counter) WithContext(ctx context.Context) CounterMetric {
-	c.ctx = ctx
-	return c.CounterMetric
-}
-
-// withExemplar initializes the exemplarMetric object and sets the exemplar value.
-func (c *Counter) withExemplar(v float64) {
-	(&exemplarCounterMetric{c}).withExemplar(v)
+	return &exemplarCounterMetric{ctx: ctx, delegate: c.CounterMetric}
 }
 
 func (c *Counter) Add(v float64) {
-	c.withExemplar(v)
+	c.CounterMetric.Add(v)
 }
 
 func (c *Counter) Inc() {
-	c.withExemplar(1)
+	c.CounterMetric.Add(1)
 }
 
-// withExemplar attaches an exemplar to the metric.
-func (e *exemplarCounterMetric) withExemplar(v float64) {
-	if m, ok := e.CounterMetric.(prometheus.ExemplarAdder); ok {
+// withExemplar is a no-op for Counter since exemplars are handled via WithContext
+func (c *Counter) withExemplar(v float64) {
```

Comment:
This is pretty confusing... when is this called on Counter directly? Can we drop it, rather than have it do nothing related to exemplars?

Reply:
I think previously we had that as a way to ensure all calls to `Counter.Add()` implicitly recorded the exemplar labels - but that was when we had context in the Counter struct. This is now ensured for Counters that first add a context and then record a value via Counter.WithContext(ctx).Add(value)